Source/carthage/Archive.swift

```diff
@@ -103,6 +103,8 @@ public struct ArchiveCommand: CommandType {
 						.map { ($0 as NSString).lastPathComponent }
 						.filter { $0.hasSuffix(".framework") }
 
+					let relativePaths = paths.map { $0.stringByReplacingOccurrencesOfString(options.directoryPath, withString: ".") }
```

Comment:
This makes it so that we convert the relative paths to absolute paths above and then convert them back to relative paths here.

It would be better to pass the relative paths all the way through and convert them to absolute paths when necessary.